src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1523,18 +1524,56 @@ public int Compare(Symbol fst, Symbol snd)
             DiagnosticBag diagnostics,
             bool suppressUseSiteDiagnostics,
             out bool wasError,
-            NamespaceOrTypeSymbol qualifierOpt = null,
+            NamespaceOrTypeSymbol qualifierOpt,
             LookupOptions options = default(LookupOptions))
         {
             Symbol symbol = resultSymbol(result, simpleName, arity, where, diagnostics, suppressUseSiteDiagnostics, out wasError, qualifierOpt, options);
 
-            if (symbol.Kind == SymbolKind.NamedType)
+            switch (symbol.Kind)
             {
-                CheckRuntimeSupportForSymbolAccess(where, receiverOpt: null, symbol, diagnostics);
+                case SymbolKind.NamedType:
+                    CheckRuntimeSupportForSymbolAccess(where, receiverOpt: null, symbol, diagnostics);
+
+                    if (shouldRecordUsedAssemblyReferences() && qualifierOpt?.IsType != true)
+                    {
+                        NamedTypeSymbol container = symbol.ContainingType;
+
+                        if (container is null)
```

Comment:
>if [](start = 24, length = 2)

Can you please explain why this whole if statement isn't just `Compilation.AddAssembliesUsedByTypeReference((TypeSymbol)symbol);`?

Reply:
> Can you please explain why this whole if statement isn't just Compilation.AddAssembliesUsedByTypeReference((TypeSymbol)symbol);?

I guess it could be. We would unnecessary visit type arguments for top level types, but it is probably not a big deal. I will simplify.

---
In reply to: [346973383](https://github.com/dotnet/roslyn/pull/39807#discussion_r346973383) [](ancestors = 346973383)